rpc/swap.go

```diff
@@ -252,3 +257,44 @@ func (s *SwapService) Cancel(_ *http.Request, req *CancelRequest, resp *CancelRe
 func offerIDStringToHash(s string) (types.Hash, error) {
 	return types.HexToHash(s)
 }
+
+// SuggestedExchangeRateResponse ...
+type SuggestedExchangeRateResponse struct {
+	ETHUpdatedAt time.Time           `json:"ethUpdatedAt"`
+	ETHPrice     *apd.Decimal        `json:"ethPrice"`
+	XMRUpdatedAt time.Time           `json:"xmrUpdatedAt"`
+	XMRPrice     *apd.Decimal        `json:"xmrPrice"`
+	ExchangeRate *coins.ExchangeRate `json:"exchangeRate"`
+}
+
+// SuggestedExchangeRate returns the current mainnet exchange rate, expressed as the XMR/ETH price.
+func (s *SwapService) SuggestedExchangeRate(_ *http.Request, _ *interface{}, resp *SuggestedExchangeRateResponse) error { //nolint:lll
+
+	ctx := context.Background() // TODO: There should be some parent context that would be better to use
```

Comment:
This bothered me, but I couldn't find anywhere to grab it from.

Reply:
we could set a timeout for this?

Reply:
I updated the code to pass the context in to `NewSwapService(...)`. I think that makes sense. The same context that cancels the local RPC services should also cancel requests made on behalf of those services.